src/fides/api/ops/api/v1/endpoints/storage_endpoints.py

```diff
@@ -278,12 +290,214 @@ def delete_config_by_key(
+        incoming_data["key"] = existing_default.key
+    else:
+        # set a name for our config if we're creating a new default
+        incoming_data["name"] = default_storage_config_name(
+            incoming_storage_config.type  # type: ignore
+        )
+
+    # since we're setting the default storage config for the given type, `is_default` MUST be set to `True`
+    incoming_data["is_default"] = True
+
+    try:
+        storage_config = StorageConfig.create_or_update(db=db, data=incoming_data)
+        return storage_config
+    except KeyOrNameAlreadyExists as exc:
+        logger.warning(
+            "Create/update failed for default config update for storage type {}: {}",
+            incoming_storage_config.type,
+            exc,
+        )
+        raise HTTPException(status_code=HTTP_400_BAD_REQUEST, detail=exc.args[0])
+    except Exception as exc:
+        logger.warning(
+            "Create/update failed for default config update for storage type {}: {}",
+            incoming_storage_config.type,
+            Pii(str(exc)),
+        )
+        raise HTTPException(
+            status_code=HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Error creating or updating storage config.",
+        )
+
+
+@router.put(
+    STORAGE_DEFAULT_SECRETS,
+    status_code=HTTP_200_OK,
+    dependencies=[Security(verify_oauth_client, scopes=[STORAGE_CREATE_OR_UPDATE])],
+    response_model=TestStatusMessage,
+)
+def put_default_config_secrets(
+    storage_type: StorageType,
+    *,
+    db: Session = Depends(deps.get_db),
+    unvalidated_storage_secrets: possible_storage_secrets,
+    verify: Optional[bool] = True,
+) -> TestStatusMessage:
+    """
+    Add or update secrets for the default storage config of the given type
+    """
+    logger.info("Finding default config of storage type '{}'", storage_type.value)
+    storage_config = get_default_storage_config_by_type(db, storage_type)
+    if not storage_config:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=f"No default configuration found for storage type {storage_type.value}.",
+        )
+    try:
+        secrets_schema = get_schema_for_secrets(
+            storage_type=storage_config.type,
+            secrets=unvalidated_storage_secrets,
+        )
+    except KeyError as exc:
+        raise HTTPException(
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=exc.args[0],
+        )
+    except ValueError as exc:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=exc.args[0],
+        )
+
+    logger.info(
+        "Updating storage config secrets for default config of storage type '{}'",
+        storage_type.value,
+    )
+    try:
+        storage_config.set_secrets(db=db, storage_secrets=secrets_schema.dict())  # type: ignore
+    except ValueError as exc:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
```

Comment:
i can't figure out a way to actually hit this code in any test, the `ValueError` coming from `set_secrets` seems totally unreachable. so i'm going to remove this try/except block.